### PR TITLE
Changed providers.ts to async/await

### PR DESCRIPTION
**Pull request by knowledgeReignsSupreme**
Changed the syntax of providers.ts controller from promise-based to async/await

### CODE REVIEW

**Comment by davidmeirlevy** [comment]
It also includes the changes of the templates.

File: server/controllers/providers.ts [hunk 1]
```diff
@@ -1,90 +1,105 @@
...
+    req.provider.metadata = body.metadata;
   }
   if (body.access) {
-    req.provider.access = body.access
+    req.provider.access = body.access;
+  }
+
+  try {
+    if (newSecretSetter) {
+      await newSecretSetter();
```
Comment (davidmeirlevy) [inline server/controllers/providers.ts:76]: Where did this function came from?

**Review by davidmeirlevy — CHANGES_REQUESTED** [review]
It looks like you're using a function that not exist..

File: server/controllers/providers.ts (hunk 1 above)
Comment (knowledgeReignsSupreme) [reply server/controllers/providers.ts:76]: Oh my. It's supposed to be:
const newSecretSetter = setSecret(...)

If (newSecretSetter){
  newSecretSetter()
}

Comment (knowledgeReignsSupreme) [reply server/controllers/providers.ts]: When replacing "promise = setSecret(..); promise.then()..." I had to think of another way to not make an async operation while looking at the req.body data, so I set a boolean named "shouldUpdateSecret" and if it's true, the said async operation will execute before updating the provider.

**Review by davidmeirlevy — APPROVED** [review]

**Review by davidmeirlevy — APPROVED** [review]

**Pull request merged**